lib/ansible/modules/pip.py

```diff
@@ -350,24 +351,56 @@ def _get_cmd_options(module, cmd):
     return cmd_options
 
 
-def _get_packages(module, pip, chdir):
-    '''Return results of pip command to get packages.'''
+def _get_packages(module, pip, chdir, env):
+    '''Return the installed packages as a string, one package per line, like:
+    black==20.8b1
+    filelock==3.0.12
+    jedi==0.17.2
+    '''
     # Try 'pip list' command first.
     command = '%s list --format=freeze' % pip
     lang_env = {'LANG': 'C', 'LC_ALL': 'C', 'LC_MESSAGES': 'C'}
     rc, out, err = module.run_command(command, cwd=chdir, environ_update=lang_env)
 
-    # If there was an error (pip version too old) then use 'pip freeze'.
+    # If there was an error (maybe pip version < 1.3) try 'pip freeze'.
     if rc != 0:
         command = '%s freeze' % pip
         rc, out, err = module.run_command(command, cwd=chdir)
         if rc != 0:
             _fail(module, command, out, err)
-
+        # pip freeze does not list setuptools or pip in its output
+        # So we need to get those via a specialcase
+        for pkg in ('setuptools', 'pip'):
+            formatted_dep = _get_package_info(module, pkg, env)
+            if formatted_dep is not None:
+                out += '%s\n' % formatted_dep
+
+    # Clean output for pip warnings
+    out = "\n".join([line for line in out.split('\n')
+                     if not line.startswith('You are using')
+                     and not line.startswith('You should consider')])
```

Comment:
you shouldn't need to create an intermediate in-memory list here:
```suggestion
    out = "\n".join(line for line in out.split('\n')
                    if not line.startswith('You are using')
                    and not line.startswith('You should consider'))
```